### DISCUSSION TITLE
Creating and linking two table entries in one form

### DISCUSSION THREAD

**Discussion by michal-81** [Budibase/budibase#8087] [Help] [answered]
Hello everyone.
Sorry for any grammar mistakes but English is not my mother tongue.

I have 2 tables: reservations and reservation_history.
It is One (reservations) to Many (reservation_history) relationship.
When I create new reservation row I would like to store additional information in reservation_history table (who created it, status etc).
I defined second 'Save Row' action as in doc example  [https://docs.budibase.com/docs/referencing](https://docs.budibase.com/docs/referencing) but I'm getting error 'Incorrect integer value' for foreign key column when saving the form.

I tried Handlebars 

![Budibase-HB](https://user-images.githubusercontent.com/114871390/193464721-e0aa21ac-db29-4465-a7d4-c263b67a9243.png)

and Javascript approach 
![Budibase-JS](https://user-images.githubusercontent.com/114871390/193464744-bcc4534c-36b6-46fc-a06b-b6c2573f1c79.png)

but both give me the same error.

![Budibase-error](https://user-images.githubusercontent.com/114871390/193464778-d7f3a765-bf38-4683-98e0-aa417b663853.png)

It looks like row ID from parent table (31 in this example) is escaped so the value no longer is treated as integer.

Could anyone help me to set it up correctly? 

Thank you in advance.
Michal

**Comment by cherrycharan** [comment]
In the table **reservation_history** what is the data type for the column _**fk_reservations_reservation_history**_?

**Reply by michal-81** [reply]
Hi @cherrycharan. In Budibase it is set as Number

![Budibase-fk-definition](https://user-images.githubusercontent.com/114871390/193620187-1362fd2b-2352-4c87-b134-4a10cb245a8a.png)

 and in MySQL as Int

![Budibase-fk-definition-DB](https://user-images.githubusercontent.com/114871390/193620491-51acda84-fe31-4a82-913b-fbe3f49a7986.png)

**Comment by melohagan** [comment, marked as answer]
Hi @michal-81 

It looks like you'll need to decode the id unfortunately. Should be able to do it with this JavaScript binding for the fk column:

```js
decodeURI($("Action 2.Saved row")._id).match(/\[(.*?)\]/)[1]
```

**Reply by michal-81** [reply]
Thank you @melohagan. It worked :) but should this behaviour be considered a bug?

**Reply by melohagan** [reply]
It's certainly not ideal! There is a related bug here: https://github.com/Budibase/budibase/issues/6194
